docs/pages/debugging/devtools-plugins.mdx

```diff
@@ -0,0 +1,201 @@
+---
+title: Dev Tools Plugins
+description: Learn about using dev tools plugins to inspect and debug your Expo Project.
+---
+
+import { Tab, Tabs } from '~/ui/components/Tabs';
+import { Terminal } from '~/ui/components/Snippet';
+
+> Available with SDK 50 and above.
+
+Dev tools plugins are tools available in your local development environment that help you to debug your app. They typically consist of a small amount of code your add to project that enables two-way communication between your app and an external Chrome window, which can then display tools for inspecting your app, triggering certain behaviors for testing, and more. Dev tools plugins are similar to Flipper plugins, but are available in Expo Go and Development Builds without adding any native modules or config plugins to your project.
+
+## Adding a dev tools plugin to your project
+
+Dev tools plugins are installed as packages in your project. A plugin will typically have a small amount of connecting code that you add to your app in order to enable two-way communication between your app and the plugin. This code will generally be invoked in your app's root component, especially if the plugin is inspecting a module that is available for the entire time your app is running. It is recommended that you wrap this code in a conditional that only runs in development mode, so that it is not included in production builds of your app:
+
+```jsx
+if (__DEV__) {
+  // initialize your DevTools plugin here
+}
```

Comment:
in the newer implementation, i'm lean toward to use the convention:

```ts
export let useReactNavigationDevTools: typeof import('./useReactNavigationDevTools').useReactNavigationDevTools;

// @ts-ignore process.env.NODE_ENV is defined by metro transform plugins
if (process.env.NODE_ENV !== 'production') {
  useReactNavigationDevTools = require('./useReactNavigationDevTools').useReactNavigationDevTools;
} else {
  useReactNavigationDevTools = () => {};
}
```

that should have better tree-shaking optimization.